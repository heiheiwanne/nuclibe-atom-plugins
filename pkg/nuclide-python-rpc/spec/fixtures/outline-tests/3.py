# Copyright (c) 2015-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the license found in the LICENSE file in
# the root directory of this source tree.


class Test3:

    def __init__(self):
        print('hi')

    @fakedecorator
    def hello(self, name):
        print('hello' + name)

    @property
    def hey(self):
        return 'hi'

    @staticmethod
    def meth():
        pass
