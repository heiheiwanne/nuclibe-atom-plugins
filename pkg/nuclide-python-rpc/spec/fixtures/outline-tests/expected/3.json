[
  {
    "kind": "class",
    "end": {
      "column": 0,
      "line": 24
    },
    "name": "Test3",
    "start": {
      "column": 0,
      "line": 8
    },
    "params": [],
    "children": [
      {
        "kind": "function",
        "end": {
          "column": 4,
          "line": 13
        },
        "name": "__init__",
        "start": {
          "column": 4,
          "line": 10
        },
        "params": [
          "self"
        ],
        "children": []
      },
      {
        "kind": "function",
        "end": {
          "column": 4,
          "line": 17
        },
        "name": "hello",
        "start": {
          "column": 4,
          "line": 14
        },
        "params": [
          "self",
          "name"
        ],
        "children": []
      },
      {
        "kind": "function",
        "end": {
          "column": 4,
          "line": 21
        },
        "name": "hey",
        "start": {
          "column": 4,
          "line": 18
        },
        "params": [
          "self"
        ],
        "children": []
      },
      {
        "kind": "function",
        "end": {
          "column": 0,
          "line": 24
        },
        "name": "meth",
        "start": {
          "column": 4,
          "line": 22
        },
        "params": [
        ],
        "children": []
      }
    ]
  }
]
